services/checks/checks.go

```diff
@@ -565,46 +574,83 @@ type sttCheckResult struct {
 	result    check.Result
 }
 
-func (s *Service) processResults(ctx context.Context, check check.Check, target target, resID string) ([]sttCheckResult, error) {
+// StarlarkScriptData represents the data we need to pass to the binary to run starlark scripts.
+type StarlarkScriptData struct {
+	CheckName    string                   `json:"name"`
+	Script       string                   `json:"script"`
+	CheckVersion uint32                   `json:"version"`
+	ScriptInput  []map[string]interface{} `json:"input"`
+}
+
+func (s *Service) processResults(ctx context.Context, sttCheck check.Check, target target, resID string) ([]sttCheckResult, error) {
 	nCtx, cancel := context.WithTimeout(ctx, resultTimeout)
 	r, err := s.waitForResult(nCtx, resID)
 	cancel()
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to get action result")
 	}
 
-	funcs, err := getFuncsForVersion(check.Version)
+	l := s.l.WithFields(logrus.Fields{
+		"name":       sttCheck.Name,
+		"id":         resID,
+		"service_id": target.serviceID,
+	})
+
+	input := &StarlarkScriptData{
+		CheckName:    sttCheck.Name,
+		Script:       sttCheck.Script,
+		CheckVersion: sttCheck.Version,
+		ScriptInput:  r,
+	}
+
+	cmdCtx, cancel := context.WithTimeout(ctx, scriptTimeout)
+	defer cancel()
+
+	process := exec.CommandContext(cmdCtx, "pmm-managed-starlark")
+	pdeathsig.Set(process, syscall.SIGTERM)
+	err = unix.Setrlimit(unix.RLIMIT_CPU, &unix.Rlimit{Cur: cpuLimit, Max: cpuLimit})
```

Comment:
You do not use `process` there, so this line sets limit to pmm-managed binary, not pmm-managed-starlark